MUXSDKStats/MUXSDKStats/MUXSDKCustomerPlayerDataStore.m

```diff
@@ -0,0 +1,32 @@
+//
+//  MUXSDKCustomerPlayerDataStore.m
+//  MUXSDKStats
+//
+//  Created by Nidhi Kulkarni on 2/3/20.
+//  Copyright © 2020 Mux, Inc. All rights reserved.
+//
+
+#import "MUXSDKCustomerPlayerDataStore.h"
+
+@interface MUXSDKCustomerPlayerDataStore()
+@property (nonatomic, retain) NSMutableDictionary *customerPlayerDatas;
+@end
+@implementation MUXSDKCustomerPlayerDataStore
+
+- (id) init {
+    self = [super init];
+    if (self) {
+        self.customerPlayerDatas = [[NSMutableDictionary alloc] init];
+    }
+    return(self);
+}
+
+- (void) setPlayerData:(nonnull MUXSDKCustomerPlayerData *)playerData forPlayerName:(nonnull NSString *)name {
+    [self.customerPlayerDatas setValue:playerData forKey:name];
+}
+
+- (MUXSDKCustomerPlayerData *_Nullable) playerDataForPlayerName:(nonnull NSString *)name {
+    return [self.customerPlayerDatas valueForKey:name];
+}
```

Comment:
>MUXSDKCustomerPlayerDataStore is being used as an in-memory cache to save playerData for a particular player.

That's right.

>We were doing this before, with a dictionary in MUXSDKStats

In your branch we were using a dictionary. I was doing essentially the same thing to fix this bug, but defined a separate class and defined a protocol so that different components that need the cached metadata could reference this protocol. It also makes it easier to write tests. I do not think we are (in the current `master`) caching customer player data in the `MUXSDKStats` layer after we dispatch the data event to `MuxCore`